src/Control/Lens/Traversal.hs

```diff
@@ -820,6 +701,74 @@ unconsWithDefault d []     = (d,[])
 unconsWithDefault _ (x:xs) = (x,xs)
 {-# INLINE unconsWithDefault #-}
 
+
+-------------------------------------------------------------------------------
+-- Holes
+-------------------------------------------------------------------------------
+
+-- | The one-level version of 'Control.Lens.Plated.contextsOf'. This extracts a
+-- list of the immediate children according to a given 'Traversal' as editable
+-- contexts.
+--
+-- Given a context you can use 'Control.Comonad.Store.Class.pos' to see the
+-- values, 'Control.Comonad.Store.Class.peek' at what the structure would be
+-- like with an edited result, or simply 'extract' the original structure.
+--
+-- @
+-- propChildren l x = 'toListOf' l x '==' 'map' 'Control.Comonad.Store.Class.pos' ('holesOf' l x)
+-- propId l x = 'all' ('==' x) ['extract' w | w <- 'holesOf' l x]
+-- @
+--
+-- @
+-- 'holesOf' :: 'Iso'' s a                -> s -> ['Pretext'' (->) a s]
+-- 'holesOf' :: 'Lens'' s a               -> s -> ['Pretext'' (->) a s]
+-- 'holesOf' :: 'Traversal'' s a          -> s -> ['Pretext'' (->) a s]
+-- 'holesOf' :: 'IndexedLens'' i s a      -> s -> ['Pretext'' ('Indexed' i) a s]
+-- 'holesOf' :: 'IndexedTraversal'' i s a -> s -> ['Pretext'' ('Indexed' i) a s]
+-- @
+holesOf :: Conjoined p
+        => Over p (Bazaar p a a) s t a a -> s -> [Pretext p a a t]
+holesOf f xs = flip appEndo [] . fst $
+  runHoles (runBazaar (f sell xs) (cotabulate oneb)) id
+{-# INLINE holesOf #-}
+-- We are very careful to share as much structure as possible among
+-- the results (in the common case where the traversal allows for such).
+-- Note in particular the recursive knot in the implementation of <*>
+-- for Holes. This sharing magic was inspired by Noah "Rampion" Easterly's
+-- implementation of a related holes function: see
+-- https://stackoverflow.com/a/49001904/1477667.
+
+newtype Holes p a x = Holes
+  { runHoles :: forall t. (x -> t) -> (Endo [Pretext p a a t], x) }
+
+oneb :: forall p a. (Corepresentable p, Category p) => Corep p a -> Holes p a a
+oneb x = Holes $ \xt -> ( Endo (fmap xt (cosieve sell x) :)
+                        , cosieve (id :: p a a) x)
+
```

Comment:
One question: do we need quite so much laziness in the instances? Can we match strictly on the pair in `fmap`, and the first of the two in each of `(<*>)` and `liftA2`? It seems likely, since we're making a list, but this stuff is a bit confusing!